controlplane/src/core/repositories/OrganizationRepository.ts

```diff
@@ -597,4 +605,246 @@ export class OrganizationRepository {
+              name: input.name,
+              type: 'slack',
+              events: input.events,
+            })
+            .returning()
+            .execute();
+
+          const slackIntegrationConfig = await db
+            .insert(slackIntegrationConfigs)
+            .values({
+              integrationId: createSlackIntegrationResult[0].id,
+              endpoint: input.endpoint,
+            })
+            .returning()
+            .execute();
+
+          for (const eventMeta of input.eventsMeta) {
+            switch (eventMeta.meta.case) {
+              case 'federatedGraphSchemaUpdated': {
+                const ids = eventMeta.meta.value.graphIds;
+                await db.insert(slackIntegrationEventConfigs).values({
+                  event: eventMeta.meta.case,
+                  slackIntegrationConfigId: slackIntegrationConfig[0].id,
+                  federatedGraphIds: ids || [],
+                });
+                break;
+              }
+            }
+          }
+        }
+      }
+    });
+  }
+
+  public async getIntegrationByName(organizationId: string, integrationName: string): Promise<Integration | undefined> {
+    const res = await this.db.query.organizationIntegrations.findMany({
+      where: and(
+        eq(organizationIntegrations.organizationId, organizationId),
+        eq(organizationIntegrations.name, integrationName),
+      ),
+    });
+
+    if (res.length === 0) {
+      return undefined;
+    }
+
+    const orgIntegrations: Integration[] = [];
+
+    for (const r of res) {
+      switch (r.type) {
+        case 'slack': {
+          const slackIntegrationConfig = await this.db.query.slackIntegrationConfigs.findFirst({
+            where: eq(slackIntegrationConfigs.integrationId, r.id),
+          });
+          if (!slackIntegrationConfig) {
+            continue;
+          }
+
+          const config: PartialMessage<IntegrationConfig> = {
+            type: IntegrationType.SLACK,
+            config: {
+              case: 'slackIntegrationConfig',
+              value: {
+                endpoint: slackIntegrationConfig.endpoint,
+              },
+            },
+          };
+
+          const integrationEventConfigs = await this.db.query.slackIntegrationEventConfigs.findMany({
+            where: eq(slackIntegrationEventConfigs.slackIntegrationConfigId, slackIntegrationConfig.id),
+          });
+
+          orgIntegrations.push({
+            id: r.id,
+            name: r.name,
+            type: r.type,
+            events: r.events || [],
+            integrationConfig: config,
+            eventsMeta: integrationEventConfigs.map((ic) => ({
+              eventName: OrganizationEventName.FEDERATED_GRAPH_SCHEMA_UPDATED,
+              meta: {
+                case: 'federatedGraphSchemaUpdated',
+                value: {
+                  graphIds: ic.federatedGraphIds,
+                },
+              },
+            })),
+          } as Integration);
+        }
+      }
```

Comment:
Throw on invalid type.

Reply:
Didnt get you

Reply:
Define a default case and throw an error. It should never happen.

Reply:
Got it. Done